homeassistant/components/airos/sensor.py

```diff
@@ -0,0 +1,152 @@
+"""AirOS Sensor component for Home Assistant."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+
+from airos.airos8data import NetRole, WirelessMode
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS,
+    UnitOfDataRate,
+    UnitOfFrequency,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .coordinator import AirOSConfigEntry, AirOSData, AirOSDataUpdateCoordinator
+from .entity import AirOSEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class AirOSSensorEntityDescription(SensorEntityDescription):
+    """Describe an AirOS sensor."""
+
+    value_fn: Callable[[AirOSData], StateType]
+
+
+SENSORS: tuple[AirOSSensorEntityDescription, ...] = (
+    AirOSSensorEntityDescription(
+        key="host_cpuload",
+        translation_key="host_cpuload",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.host.cpuload,
+        entity_registry_enabled_default=False,
+    ),
+    AirOSSensorEntityDescription(
+        key="host_netrole",
+        translation_key="host_netrole",
+        value_fn=lambda data: data.host.netrole.value
+        if isinstance(data.host.netrole, NetRole)
+        else data.host.netrole,
```

Comment:
This is an entity with limited values, this can be an enum entity where we provide the available options in `options`. If a value is not in `options`, it would raise. I think you can add some logic in the library that checks if we got a new value that we don't know yet and ask the user to raise an issue https://github.com/joostlek/python-open-router/blob/main/src/python_open_router/models.py#L129. This is an efficient way to get data from users imo

Reply:
Said logic is present presenting a warning logged once (per instantiation) but written in a way not to break it in the early conception without having user data present. I only took the post serialisation route (https://github.com/CoMPaTech/python-airos/blob/main/airos/airos8data.py), will rewrite to pre as suggested.